src/main/java/com/bookstore/bookstoreapi/data/service/WishlistService.java

```diff
@@ -37,4 +40,29 @@ public Optional<Wishlist> createWishlist(String wishlistName, Long userID) {
 
     return Optional.of(newWishlist);
   }
+
+  public Optional<Wishlist> addBookToWishlist(Long bookID, Long wishlistID) {
+
+    Optional<Book> bookOptional = bookRepository.findById(bookID);
+    Optional<Wishlist> wishlistOptional = wishlistRepository.findById(wishlistID);
+
+    if (bookOptional.isEmpty() || wishlistOptional.isEmpty()) {
+      return Optional.empty();
+    }
+
+    Book book = bookOptional.get();
+    Wishlist wishlist = wishlistOptional.get();
+
+    if (wishlist.getBooks().contains(book)) {
+      return Optional.empty();
+    }
+
+    wishlist.getBooks().add(book);
+    book.getWishlists().add(wishlist);
+
+    wishlistRepository.save(wishlist);
+    bookRepository.save(book);
```

Comment:
saving it both ways for protection. book is the owner but it does not hurt too much.